internal/app/init/pkg/network/network.go

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package network
+
+import (
+	"errors"
+	"log"
+	"time"
+
+	"github.com/autonomy/dhcp/dhcpv4"
+	"github.com/autonomy/dhcp/dhcpv4/client4"
+	"github.com/autonomy/dhcp/netboot"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+// Setup creates the network.
+func Setup(platform string) (err error) {
+	//ifup lo
+	ifname := "lo"
+	link, err := netlink.LinkByName(ifname)
+	if err != nil {
+		log.Println(err)
```

Comment:
These errors should bubble up and be spit out in `main.go`. Was that not the case? Curious if the recovery logic is broken in `main.tf`.

Reply:
I feel like I was running into the same case with the userdata stuff in this PR https://github.com/autonomy/talos/pull/398/files. Yeah, may be worth verifying that the recovery function is outputting things as necessary.